mobile/library/objective-c/EnvoyEngineImpl.mm

```diff
@@ -609,4 +607,16 @@ - (void)terminateNotification:(NSNotification *)notification {
   terminate_engine(_engineHandle, /* release */ false);
 }
 
+- (void)logException:(NSException *)exception {
+  NSLog(@"[Envoy] exception caught: %@", exception);
+
+  NSString *message = [NSString stringWithFormat:@"%@;%@;%@",
+                       exception.name, exception.reason, exception.callStackSymbols.description];
+  ENVOY_LOG_EVENT_TO_LOGGER(Envoy::Logger::Registry::getLog(Envoy::Logger::Id::misc),
+                            error, "handled_cxx_exception",
+                            [message UTF8String]);
+
+  [NSNotificationCenter.defaultCenter postNotificationName:@"EnvoyError" object:self];
```

Comment:
Technically a breaking change, but it seems like it would be significantly more useful to attach the exception to the notification:

```suggestion
  [NSNotificationCenter.defaultCenter postNotificationName:@"EnvoyError" object:exception];
```